### ISSUE TITLE
[BUG] Weird engine sound

### ISSUE THREAD

**Issue by dobriyprop** [ACF-Team/ACF-3#392]
**Short Description**

I suspect after https://github.com/ACF-Team/ACF-3/pull/378 update to dev branch engine sound started to act weird. On the dev branch, it sounds like engine sounds are updating at a much slower rate than on master, almost sounding like they are being interpolated rather than directly taking the engine's RPM.

**Branch and Version**

dev branch version bbe741d

**How to Reproduce (Optional)**

Spawn an engine with light flywheel and hear for yourself

**Videos**

master branch

https://github.com/ACF-Team/ACF-3/assets/18686703/cc904d6f-f730-40cc-b8a3-db13edd4c80d

dev branch

https://github.com/ACF-Team/ACF-3/assets/18686703/26f81e3d-ec81-4f61-b2b0-b2bf10d51d88

**Comment by thecraftianman** [comment]
Do you still have this issue on the latest commit to dev as well? This may be the volume issue that I resolved in 8a3bd81 after the commit that you listed.

**Comment by dobriyprop** [comment]
Sorry. I specified the wrong commit. Didn't notice that gtihub page switched back to master. The commit is [15007b6](https://github.com/ACF-Team/ACF-3/commit/15007b60fd48adcd660c9c48578575aa5968f5e8). To be completely sure i redownloaded whole repo right now, switched to dev branch and tested once more - the issue is still present, unfortunately.

```
PS D:\SteamLibrary\steamapps\common\GarrysMod\garrysmod\addons> git clone https://github.com/ACF-Team/ACF-3.git
Cloning into 'ACF-3'...
remote: Enumerating objects: 20930, done.
remote: Counting objects: 100% (3243/3243), done.
remote: Compressing objects: 100% (1188/1188), done.
remote: Total 20930 (delta 2089), reused 3034 (delta 1928), pack-reused 17687
Receiving objects: 100% (20930/20930), 795.22 MiB | 3.40 MiB/s, done.
Resolving deltas: 100% (11150/11150), done.
Updating files: 100% (1137/1137), done.
PS D:\SteamLibrary\steamapps\common\GarrysMod\garrysmod\addons> git checkout dev
fatal: not a git repository (or any of the parent directories): .git
PS D:\SteamLibrary\steamapps\common\GarrysMod\garrysmod\addons> cd .\ACF-3\
PS D:\SteamLibrary\steamapps\common\GarrysMod\garrysmod\addons\ACF-3> git checkout dev
Switched to a new branch 'dev'
branch 'dev' set up to track 'origin/dev'.
PS D:\SteamLibrary\steamapps\common\GarrysMod\garrysmod\addons\ACF-3> git pull
Already up to date.
```
![изображение](https://github.com/ACF-Team/ACF-3/assets/18686703/a2467481-fcac-465c-aa1d-2527cbb8fdcc)

**Comment by thecraftianman** [comment]
Just pushed something that should resolve this to dev; can you confirm that it works for your engine setup too?

**Comment by dobriyprop** [comment]
The pitch of engine sound still updates too infrequently. Engine sounds too lazy. Is there anything that could be done about that?

https://github.com/ACF-Team/ACF-3/assets/18686703/6663cdf2-fa6a-4cad-859d-9add9235cb1e

**Comment by thecraftianman** [comment]
How about now?

**Comment by dobriyprop** [comment]
Much better. Engine sounds more responsive now. Though refresh rate is not every tick like before but i guess that would put too much strain on networking, right?

[2024-03-13_23-34.webm](https://github.com/ACF-Team/ACF-3/assets/18686703/07456e3b-eee7-4327-9e99-65bb6080d42b)

**Comment by thecraftianman** [comment]
Yeah, there's currently a limit on how quickly sounds can be updated to avoid having to send as much network traffic to clients, which is partly the cause of this issue in the first place. If things are no longer a problem now, though, I'll go ahead and close this issue. Thanks for your help testing!

**Comment by dobriyprop** [comment]
We found another issue with sounds here.
https://github.com/ACF-Team/ACF-3/blob/60cbefce2bd82d9e0f6997379b566c44ea2f764d/lua/acf/core/utilities/sounds/sounds_sv.lua#L53
Updates to pitch and volume being sent by reliable channel. If ping between server and client is big enough, this will cause additional delays in updating pitch and volume of sounds due to client will have to send confirmation receipt before listening for another chunk of data from the server. Maybe for a single run sounds like gun shots that's fine but for engine sounds it causes audible choppiness on high ping servers.

**Comment by dobriyprop** [comment]
If you're testing it in singleplayer try it with `net_fakelag 50` or higher and listen to the sounds with sending data by reliable and unreliable channels.

Here how it sounds with `net_fakelag 0` and `net_fakelag 50` on Git-dev-cc340f3 version with no fixes for that.

[](https://github.com/ACF-Team/ACF-3/assets/18686703/c66cecb8-fb2b-4212-9a9a-937cf0ab71e9)

You can hear audible choppiness when ping reaches 150+ ms. This happens because server waits until client confirms that it received previous chunk of data before sending the next one. And if the time taken to server send the data and receive a confirmation of receipt from the client is bigger than server's rate at which it tries to send data (currently 50ms) it causes it to skip sending next data update to the client causing choppiness.

And here how it sounds with my custom engine sound Starfall chip that sends data by unreliable channel with `net_fakelag 0` and `net_fakelag 50`

[](https://github.com/ACF-Team/ACF-3/assets/18686703/682920fb-4fb9-4594-98fc-80d7bb77ee36)

[engine_sound.txt](https://github.com/ACF-Team/ACF-3/files/14728953/engine_sound.txt)

There's absolutely no choppiness, just a reaction delay between me pressing throttle key and rpm starting to increase is rpm.

**Comment by dobriyprop** [comment]
@thecraftianman, I might misunderstood your previous message. Rn I pulled latest dev version and tested it out - engine sound is no longer choppy with net_fakelag 50+. Also asked an admin of a server i usually play on to update ACF and I tested it out there - it sounds just as if there is no ping. I think it's finally fixed. Thank you!

[2024-03-23_20-14.webm](https://github.com/ACF-Team/ACF-3/assets/18686703/97ae0879-c4ed-429d-9037-24e73d1fcc1b)
